vshard/util.lua

```diff
@@ -286,6 +286,14 @@ local function fiber_is_self_canceled()
     return not pcall(fiber.testcancel)
 end
 
+--
+-- Check if connection's fiber was explicitly killed
+--
+local is_conn_fiber_killed = function(conn)
+    local msg = 'fiber is cancelled'
+    return conn.error and (conn.error == msg or conn.error.message == msg)
```

Comment:
The solution, you proposed, will work everywhere except the versions range, which starts from the time net.box was ported to C. The state of the connection in these versions is `error_reconnect` and there's no `_fiber` in it. 

The only way we can figure out that the fiber is dead is checking error message, unfortunately. Do we want that? False-positive triggering of reinitialization of connection seems to be a better option rather than just saying that reestablishing won't work on these version.  

Speaking of the earlier versions, we need nothing but checking if `conn.state` is `error`.